### PR TITLE
Refactor country group, currency and country logic

### PR DESCRIPTION
**Pull request by svillafe**
## Why are you doing this?

To be more consistent and therefore reduce the complexity of the codebase.

[**Trello Card**](https://trello.com/c/6pCA5NNu/257-refactor-country-group-logic-in-support-frontend)

## Changes

* Created the `internationalisation` object inside `CommonState`
* Avoid confusion by having only ids inside `internationalisation`.
* Propagated this changes to all the codebase.

### CODE REVIEW

File: assets/pages/digital-subscription-landing/components/priceCtaContainer.jsx [hunk 1]
```diff
@@ -9,21 +9,22 @@ import PriceCta from 'components/priceCta/priceCta';
...
-  const price = digitalSubPrices[state.common.countryGroup].toFixed(2);
+  const { countryGroupId } = state.common.internationalisation;
+  const price = digitalSubPrices[countryGroupId].toFixed(2);
 
   return {
     ctaText: 'Start a 14 day free trial',
     url: getDigitalCheckout(
       state.common.referrerAcquisitionData,
-      state.common.countryGroup,
+      state.common.internationalisation.countryGroupId,
```
Comment (Ap0c) [inline assets/pages/digital-subscription-landing/components/priceCtaContainer.jsx]: You've got this from line 18 haven't you?

File: assets/pages/regular-contributions/components/contributionsCheckoutContainer.jsx [hunk 2]
```diff
@@ -20,8 +20,8 @@ function mapStateToProps(state: State) {
 
   return {
     amount: state.page.regularContrib.amount,
-    currency: state.common.currency,
-    country: state.common.country,
+    currencyId: state.common.internationalisation.currencyId,
+    country: state.common.internationalisation.countryId,
```
Comment (Ap0c) [inline assets/pages/regular-contributions/components/contributionsCheckoutContainer.jsx]: `countryId`?

File: assets/pages/regular-contributions/components/formFields.jsx [hunk 3]
```diff
@@ -46,8 +46,8 @@ function mapStateToProps(state) {
   return {
     firstName: state.page.user.firstName,
     lastName: state.page.user.lastName,
-    countryGroup: state.common.countryGroup,
-    country: state.common.country,
+    countryGroup: state.common.internationalisation.countryGroupId,
+    country: state.common.internationalisation.countryId,
```
Comment (Ap0c) [inline assets/pages/regular-contributions/components/formFields.jsx]: `countryGroupId` and `countryId`?

File: assets/pages/regular-contributions/components/regularContributionsPayment.jsx [hunk 4]
```diff
@@ -170,11 +170,11 @@ function mapStateToProps(state) {
     error: state.page.regularContrib.error,
     paymentStatus: state.page.regularContrib.paymentStatus,
     amount: state.page.regularContrib.amount,
+    currencyId: state.common.internationalisation.currencyId,
     contributionType: state.page.regularContrib.contributionType,
-    currency: state.common.currency,
     regularContrib: state.page.regularContrib,
     csrf: state.page.csrf,
-    country: state.common.country,
+    country: state.common.internationalisation.countryId,
```
Comment (Ap0c) [inline assets/pages/regular-contributions/components/regularContributionsPayment.jsx]: `countryId`?

File: assets/pages/support-landing/components/contributionPaymentCtasContainer.js [hunk 5]
```diff
@@ -18,9 +18,9 @@ function mapStateToProps(state: State) {
     contributionType: state.page.selection.contributionType,
     amount: getAmount(state.page.selection),
     referrerAcquisitionData: state.common.referrerAcquisitionData,
-    country: state.common.country,
-    countryGroupId: state.common.countryGroup,
-    currency: state.common.currency,
+    country: state.common.internationalisation.countryId,
```
Comment (Ap0c) [inline assets/pages/support-landing/components/contributionPaymentCtasContainer.js]: `countryId`?

File: assets/pages/support-landing/components/contributionSelectionContainer.js [hunk 6]
```diff
@@ -16,9 +16,9 @@ import type { State } from '../supportLandingReducer';
 function mapStateToProps(state: State) {
 
   return {
-    country: state.common.country,
-    countryGroupId: state.common.countryGroup,
-    currency: state.common.currency,
+    country: state.common.internationalisation.countryId,
```
Comment (Ap0c) [inline assets/pages/support-landing/components/contributionSelectionContainer.js]: `countryId`?

**Comment by svillafe** [comment]
As discussed offline, I will not be changing props' names in this PR. This is because it is not safe to do it all in one goal because Flow is not correctly checking the containers.

File: assets/helpers/internationalisation/currency.js [hunk 7]
```diff
@@ -92,52 +92,39 @@ const spokenCurrencies = {
...
   return null;
 }
 
-function detect(countryGroup: CountryGroupId): Currency {
-  return fromQueryParameter() || fromCountryGroupId(countryGroup) || GBP;
+function detect(countryGroup: CountryGroupId): IsoCurrency {
+  return fromQueryParameter() || fromCountryGroupId(countryGroup) || 'GBP';
 }
 
 export {
```
[the remark this answers is not in the document]
Comment (svillafe) [reply assets/helpers/internationalisation/currency.js]: Fixed here: fa7af4f

File: assets/pages/digital-subscription-landing/components/priceCtaContainer.jsx (hunk 1 above)
Comment (svillafe) [reply assets/pages/digital-subscription-landing/components/priceCtaContainer.jsx]: Fixed here: c032fbc

File: assets/pages/support-landing/components/contributionSelectionContainer.js (hunk 6 above)
Comment (svillafe) [reply assets/pages/support-landing/components/contributionSelectionContainer.js]: As offline discussed, won't fix

File: assets/pages/support-landing/components/contributionPaymentCtasContainer.js (hunk 5 above)
Comment (svillafe) [reply assets/pages/support-landing/components/contributionPaymentCtasContainer.js]: As offline discussed, won't fix

File: assets/pages/regular-contributions/components/regularContributionsPayment.jsx (hunk 4 above)
Comment (svillafe) [reply assets/pages/regular-contributions/components/regularContributionsPayment.jsx]: As offline discussed, won't fix

File: assets/pages/regular-contributions/components/formFields.jsx (hunk 3 above)
Comment (svillafe) [reply assets/pages/regular-contributions/components/formFields.jsx]: As offline discussed, won't fix

File: assets/pages/regular-contributions/components/contributionsCheckoutContainer.jsx (hunk 2 above)
Comment (svillafe) [reply assets/pages/regular-contributions/components/contributionsCheckoutContainer.jsx]: As offline discussed, won't fix

File: assets/components/headers/countrySwitcherHeader/countrySwitcherHeader.jsx [hunk 8]
```diff
@@ -22,7 +22,7 @@ const CountrySwitcherHeader = (props: PropTypes) => (
     <div className="component-country-switcher-header__content">
       <CountryGroupSwitcher
         countryGroupIds={props.countryGroupIds}
-        selectedCountryGroup={props.selectedCountryGroup}
+        selectedCountryGroup={props.selectedCountryGroupId}
```
[the remark this answers is not in the document]
Comment (svillafe) [reply assets/components/headers/countrySwitcherHeader/countrySwitcherHeader.jsx]: As offline discussed, won't fix

File: assets/helpers/page/page.js [hunk 9]
```diff
@@ -39,21 +39,25 @@ import type { Action } from './pageActions';
...
   abParticipations: Participations,
   switches: Switches,
+  internationalisation: Internationalisation,
 };
 
 export type PreloadedState = {
   campaign?: $PropertyType<CommonState, 'campaign'>,
   referrerAcquisitionData?: $PropertyType<CommonState, 'referrerAcquisitionData'>,
-  country?: $PropertyType<CommonState, 'country'>,
+  country?: $PropertyType<Internationalisation, 'countryId'>,
```
[the remark this answers is not in the document]
Comment (svillafe) [reply assets/helpers/page/page.js]: Fixed here: d339f32

File: assets/components/contributionPaymentCtas/contributionPaymentCtas.jsx [hunk 10]
```diff
@@ -18,12 +18,13 @@ import { classNameWithModifiers } from 'helpers/utilities';
...
 import { addQueryParamsToURL } from 'helpers/url';
 
-import type { Currency } from 'helpers/internationalisation/currency';
+import { type IsoCurrency } from 'helpers/internationalisation/currency';
 import type { Contrib as ContributionType } from 'helpers/contributions';
 import type { Status } from 'helpers/switch';
 import type { IsoCountry } from 'helpers/internationalisation/country';
 import type { ReferrerAcquisitionData } from 'helpers/tracking/acquisitions';
 import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
+import { currencies } from 'helpers/internationalisation/currency';
```
[the remark this answers is not in the document]
Comment (svillafe) [reply assets/components/contributionPaymentCtas/contributionPaymentCtas.jsx]: Fixed here: fa7af4f

File: assets/containerisableComponents/contributionSelection/contributionSelection.jsx [hunk 11]
```diff
@@ -72,7 +72,7 @@ function ContributionSelection(props: PropTypes) {
           radios={
             getContributionAmountRadios(
               props.contributionType,
-              props.currency,
+              currencies[props.currencyId],
               props.countryGroupId,
             )}
```
[the remark this answers is not in the document]
Comment (svillafe) [reply assets/containerisableComponents/contributionSelection/contributionSelection.jsx]: Fixed here: fa7af4fa882148ed92ab73fbb1efb5d03e738197

File: assets/helpers/internationalisation/currency.js [hunk 12]
```diff
@@ -14,55 +14,36 @@ export type IsoCurrency =
...
-};
-
-export const CAD: Currency = {
-  iso: 'CAD',
-  glyph: '$',
-  extendedGlyph: 'CA$',
-};
+|};
 
 const currencies = {
```
Comment (Ap0c) [inline assets/helpers/internationalisation/currency.js]: Could you add a type for this?

```jsx
const currencies: {
  IsoCurrency: Currency,
} = {
  ...
};
```

Comment (svillafe) [reply assets/helpers/internationalisation/currency.js]: Added it here: 3259dcf

**Comment by svillafe** [comment]
Tested one-off US contribution with a credit card ✅

**Comment by svillafe** [comment]
Tested NZD recurring contribution with credit card pop up ✅ 
Tested CAD recurring contribution with PayPal ✅ 
Tested GBP recurring contribution with credit card inline ✅ 
Tested NZD one-off contribution with PayPal ✅

**Pull request merged**